rtic-sync/src/channel.rs

```diff
@@ -402,9 +442,15 @@ impl<T, const N: usize> Sender<'_, T, N> {
                 }
                 // We are not in the wait queue, no one else is waiting, and there is a free slot available.
                 else {
-                    assert!(!self.0.access(cs).freeq.is_empty());
-                    let slot = unsafe { self.0.access(cs).freeq.pop_back_unchecked() };
-                    Poll::Ready(Ok(slot))
+                    // SAFETY: `self.0.freeq` is not called recursively.
+                    unsafe {
+                        self.0.freeq(cs, |freeq| {
+                            assert!(!freeq.is_empty());
```

Comment:
Should this now be `debug_assert`?

Reply:
This one _might_ be OK. Would prefer to handle that in #1040.